frontend/src/metabase/visualizations/visualizations/ObjectDetail.jsx

```diff
@@ -25,7 +26,20 @@ const ObjectDetailProperties = {
   disableClickBehavior: true,
   settings: {
     ...columnSettings({ hidden: true }),
-    ...tableColumnSettings,
+    ...produce(tableColumnSettings, draft => {
+      draft["table.columns"].getDefault = params => {
```

Comment:
Instead of overwriting the setting, could we convert `tableColumnSettings` to a function that makes it the desired column visibility configurable?

Reply:
Thanks, done

Should I revert this change now that is not needed, or should we keep it as it's cleaner? https://github.com/metabase/metabase/pull/33828/files#diff-a23ce9e47bb10238ce74e85c2c579d47f0f6bcd4491c06cda65175ea41747e8cR57-R61